### PR TITLE
Add MU support to blueprints test helpers

### PR DESCRIPTION
**Pull request by dcyriller**
This PR unlocks the ability to do:
`emberNew({ isModuleUnification: true })`
or
`emberGenerateDestroy({ isModuleUnification: true })`

It would be especially usefull for testing blueprints in `ember.js` and `ember-data` repos + addons.

### CODE REVIEW

**Comment by dcyriller** [comment]
🤔 the tests failure doesn't seem to be related to the PR

**Comment by Turbo87** [comment]
can you add tests for that?

**Comment by dcyriller** [comment]
Sure, I'll do it. I have to figure out how to test `emberNew`.

**Comment by dcyriller** [comment]
Okay @Turbo87 I added a test for that.

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 language: node_js
 
 node_js:
+  - '10'
+  - '8'
   - '6'
-  - '4'
```
Comment (Turbo87) [inline .travis.yml]: can you extract these changes to a separate PR? if we bump the required Node versions then it's a breaking change and deserves a dedicated PR so that we can label it properly in the changelog.

**Comment by dcyriller** [comment]
I updated the PR, thanks for the review @Turbo87 

Here is the link to the CI update: https://github.com/ember-cli/ember-cli-blueprint-test-helpers/pull/155

It was required since this PR upgrades `ember-cli` to v3.2 (do not support nodejs v4 anymore).

**Comment by runspired** [comment]
Just wanted to drop by and say thanks to @dcyriller for doing this :) Looking forward to having MU blueprints for ember-data :D

File: lib/ember-new.js [hunk 2]
```diff
@@ -27,6 +27,8 @@ module.exports = function(options) {
     disableDependencyChecker: true
   };
 
+  process.env["MODULE_UNIFICATION"] = !!options.isModuleUnification;
```
Comment (rwjblue) [inline lib/ember-new.js]: Should be `EMBER_CLI_MODULE_UNIFICATION` I think 🤔...

Comment (dcyriller) [reply lib/ember-new.js]: The tests I added are failing when updating to `EMBER_CLI_MODULE_UNIFICATION`. I'll investigate why.

Comment (dcyriller) [reply lib/ember-new.js]: Okay, I get it! https://github.com/ember-cli/ember-cli/pull/7869 is still on ember-cli's beta branch. Next ember-cli release, it should be good. :)

**Comment by dcyriller** [comment]
Ok @mixonic I'll update the PR to cover these different scenarios tomorrow.

Something remains unclear to me though. It looks to me that experiments are enabled on `ember-cli`'s master branch only. I mean, I can't use them from released `ember-cli` versions. How will I be able to set `EMBER_CLI_MODULE_UNIFICATION` variable?

**Comment by dcyriller** [comment]
Taken individually, all tests are passing! 🎉 

But because they are run on the same process, they fail when run in sequence. 

Indeed, the [`experiments` object](https://github.com/ember-cli/ember-cli/blob/v3.4.0-beta.2/lib/experiments/index.js#L25) is put in cache by nodejs runtime. And when execution comes to the MU part of the tests, it fails because it resolves `experiments.MODULE_UNIFICATION` to undefined.

A solution would be to turn this object into a function. I implemented it in https://github.com/ember-cli/ember-cli/pull/7961

Does this looks good to you @rwjblue?

**Comment by dcyriller** [comment]
I manually relaunched the tests. One test suite is still failing. It's the one that generates coverage. But it's also failing on master.

**Comment by dcyriller** [comment]
If using ember-cli canary is not too much of an issue, the PR is now ready for review! All tests are passing.

**Comment by dcyriller** [comment]
An example of usage here: https://github.com/emberjs/data/pull/5558

**Comment by dcyriller** [comment]
@rwjblue @mixonic if you have time for this, your reviews are welcomed :)

This would unlock MU support for `ember-data` blueprints (+ prepare the ground to improve `ember-source` blueprints testing).

File: package.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "tmp-sync": "^1.0.0"
   },
   "devDependencies": {
-    "ember-cli": "~2.16.0",
+    "ember-cli": "github:ember-cli/ember-cli",
```
Comment (mixonic) [inline package.json]: Can this reference a specific version, or does it still need to be pinned to canary?

Comment (dcyriller) [reply package.json]: It still needs to be pinned to canary for now, specifically this ember-cli's PR https://github.com/ember-cli/ember-cli/pull/7961 (not yet on beta branch).

Comment (mixonic) [reply package.json]: Chatted with @rwjblue and he is 👍 on this, at least as a temporary solution.

Comment (dcyriller) [reply package.json]: Great! Thank you @mixonic. I'll update to a published ember-cli version when available.

Comment (dcyriller) [reply package.json]: Do you think it could be released?
